src/EnergyPlus/ZoneEquipmentManager.cc

```diff
@@ -2774,14 +2774,21 @@ namespace ZoneEquipmentManager {
 
 					// initialize sizing conditions if they have not been set (i.e., no corresponding load) to zone condition
 					if ( FinalZoneSizing( CtrlZoneNum ).ZoneTempAtCoolPeak == 0.0 ) {
-						FinalZoneSizing( CtrlZoneNum ).ZoneTempAtCoolPeak = ZoneSizing( DDNumF, CtrlZoneNum ).DesCoolSetPtSeq( TimeStepAtPeakF );
+						// issue 6006, heating coils sizing to 0 when no heating load in zone
+						if ( ZoneSizing( DDNumF, CtrlZoneNum ).DesCoolSetPtSeq.empty() ) {
+							FinalZoneSizing( CtrlZoneNum ).ZoneTempAtCoolPeak = ZoneSizing( DDNumF, CtrlZoneNum ).DesCoolSetPtSeq( TimeStepAtPeakF );
```

Comment:
If the array will never be empty then I guess you don't need the emptiness check. It will fatal error if you try to dereference the iterator from max_element and the array is empty. Maybe that is what you want.

Also, @mjwitte I'm unsure if the line will fail in Release build. I know it definitely will in Debug, but I think bounds checking doesn't happen in Release.